player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/qspatch.js

```diff
@@ -0,0 +1,379 @@
+                    } else {
+                        if (EkstepRendererAPI.isStreamingContent()) {
+                            // mobile online streaming
+                            if(path)
+                            return instance.validateUrl(EkstepRendererAPI.getBaseURL() + 'content-plugins/' + pluginId + '-' + pluginVer + '/' +path);
+                            //return org.ekstep.pluginframework.pluginManager.resolvePluginResource(pluginId, pluginVer, path);
+                        } else {
+                            // Loading content from mobile storage ( OFFLINE )
+                            return instance.validateUrl('file:///' + EkstepRendererAPI.getBaseURL() + 'content-plugins/' + pluginId + '-' + pluginVer + '/' + path);
+                        }
+                    }
+                }
+            break;
+
+            case "iconUrl":
+                pluginObj.prototype.getAudioIcon = function (path) {
+                    if (isbrowserpreview) {
+                        return instance.validateUrl(this.getAssetUrl(org.ekstep.pluginframework.pluginManager.resolvePluginResource(this._manifest.id, this._manifest.ver, path)));
+                    } else {
+                        if (EkstepRendererAPI.isStreamingContent()) {
+                            // mobile online streaming
+                            if(path)
+                            return instance.validateUrl(EkstepRendererAPI.getBaseURL() + 'content-plugins/' + this._manifest.id + '-' + this._manifest.ver + '/' + path);
+                            //return org.ekstep.pluginframework.pluginManager.resolvePluginResource(this._manifest.id, this._manifest.ver, path);
+                        } else {
+                            // Loading content from mobile storage ( OFFLINE )
+                            return instance.validateUrl('file:///' + EkstepRendererAPI.getBaseURL() + 'content-plugins/' + this._manifest.id + '-' + this._manifest.ver + '/' + path);
+                        }
+                    }
+                }
+            break;
+
+            default:
+                break;
+        }
+
+    },
+    validateUrl: function(url){
+        if(!url){
+            return
+        }
+        var regex = new RegExp("^(http|https)://", "i");
+        if(regex.test(url)){
+            var tempUrl = url.split("://")
+            if (tempUrl.length > 1){
+                var validString = tempUrl[1].split("//").join("/");
+                return [tempUrl[0], validString].join("://");
+            }
+        }else{
+            var tempUrl = url.split(":///")
+            if (tempUrl.length > 1){
+                var validString = tempUrl[1].split("//").join("/");;
+                return [tempUrl[0], validString].join(":///");
+            }
+        }
+        return url.split("//").join("/");
+    },
+    telemetryPatch: function() {
```

Comment:
Put try-catch block for this, any error while conversion of object should send data in the older format. So that it will not fail sending assess event & also will make sure application/content will not stuck because of JS error.

Reply:
Already **qspatch.telemetryPatch()** is wrapped around a try catch
**player/public/coreplugins/org.ekstep.ecmlrenderer-1.0/renderer/plugin.js** .

I am adding try catch inside **QSTelemetryLogger.logAssessEnd** function over-ride ( refer line number-161 )